python/raft-dask/raft_dask/common/comms.py

```diff
@@ -686,8 +688,75 @@ def _func_ucp_ports(client, workers):
     return client.run(_func_ucp_listener_port, workers=workers)
 
 
-def _func_worker_ranks(workers):
+def _func_worker_ranks(client):
     """
-    Builds a dictionary of { (worker_address, worker_port) : worker_rank }
+    For each worker connected to the client,
+    compute a global rank which is the sum
+    of the NVML device index and the worker rank offset.
+
+    Parameters
+    ----------
+        client (object): Dask client object.
+    """
+    ranks = client.run(_get_nvml_device_index)
+    worker_ips = [_get_worker_ip(worker_address) for worker_address in ranks]
+    worker_ip_offset_dict = _get_rank_offset_across_nodes(worker_ips)
+    return _append_rank_offset(ranks, worker_ip_offset_dict)
+
+
+def _get_nvml_device_index():
+    """
+    Return NVML device index based on environment variable
+    'CUDA_VISIBLE_DEVICES'.
+    """
+    CUDA_VISIBLE_DEVICES = os.getenv("CUDA_VISIBLE_DEVICES")
+    return nvml_device_index(0, CUDA_VISIBLE_DEVICES)
```

Comment:
Can we guarantee `CUDA_VISIBLE_DEVICES` is always set? Otherwise, I think this will break if it isn't.
(untested, needs formatting and another import, but you get the idea)
```suggestion
    CUDA_VISIBLE_DEVICES = os.getenv("CUDA_VISIBLE_DEVICES", ",".join((str(d) for d in range(dask_cuda.utils.get_gpu_count()))))
    return nvml_device_index(0, CUDA_VISIBLE_DEVICES)
```

Reply:
I think `CUDA_VISIBLE_DEVICES` is always set by `dask-cuda` 

https://github.com/rapidsai/dask-cuda/blob/bba3d3f2a1edf5b42160d8f081b676cd4e9b9a41/dask_cuda/cuda_worker.py#L207


https://github.com/rapidsai/dask-cuda/blob/bba3d3f2a1edf5b42160d8f081b676cd4e9b9a41/dask_cuda/cuda_worker.py#L233